gymnasium/envs/mujoco/humanoid_v5.py

```diff
@@ -0,0 +1,516 @@
+    [Mujoco Documentation](https://mujoco.readthedocs.io/en/latest/XMLreference.html).
+
+    **Note:** Humanoid-v4 environment no longer has the following contact forces issue.
+    If using previous Humanoid versions from v4, there have been reported issues that using a Mujoco-Py version > 2.0
+    results in the contact forces always being 0. As such we recommend to use a Mujoco-Py
+    version < 2.0 when using the Humanoid environment if you would like to report results
+    with contact forces (if contact forces are not used in your experiments, you can use
+    version > 2.0).
+
+
+    ## Rewards
+    The reward consists of three parts:
+    - *healthy_reward*:
+    Every timestep that the Humanoid is alive (see section Episode Termination for definition),
+    it gets a reward of fixed value `healthy_reward`.
+    - *forward_reward*:
+    A reward for moving forward,
+    this reward would be positive if the Humanoid moves forward (in the positive $x$ direction / in the right direction).
+    $w_{forward} \times \frac{dx}{dt}$, where
+    $dx$ is the displacement of the center of mass ($x_{after-action} - x_{before-action}$),
+    $dt$ is the time between actions, which depends on the `frame_skip` parameter (default is 5),
+    and `frametime` which is 0.001 - so the default is $dt = 5 \times 0.003 = 0.015$,
+    $w_{forward}$ is the `forward_reward_weight` (default is $1.25$).
+    The center of mass calculation is defined in the `.py` file for the Humanoid.
+    - *ctrl_cost*:
+    A negative reward to penalize the Humanoid for taking actions that are too large.
+    $w_{control} \times \\|action\\|_2^2$,
+    where $w_{control}$ is `ctrl_cost_weight` (default is $0.1$).
+    If there are *nu* actuators/controls, then the control has shape `nu x 1`.
+    - *contact_cost*:
+    A negative reward to penalize the Humanoid if the external contact forces are too large.
+    $w_{contact} \times clamp(contact\\_cost\\_range, \\|F_{contact}\\|_2^2)$, where
+    $w_{contact}$ is `contact_cost_weight` (default is $5\times10^{-7}$),
+    $F_{contact}$ are the external contact forces (see `cfrc_ext` section on observation).
+
+    The total reward returned is ***reward*** *=* *healthy_reward + forward_reward - ctrl_cost - contact_cost*
+    and `info` will also contain the individual reward terms.
+
+    Note: in `v4` the total reward returned is ***reward*** *=* *healthy_reward + forward_reward - ctrl_cost*
+
+
+    ## Starting State
+    All observations start in state
+    (0.0, 0.0,  1.4, 1.0, 0.0  ... 0.0) with a uniform noise in the range
+    of [-`reset_noise_scale`, `reset_noise_scale`] added to the positional and velocity values (values in the table)
+    for stochasticity. Note that the initial z coordinate is intentionally
+    selected to be high, thereby indicating a standing up humanoid. The initial
+    orientation is designed to make it face forward as well.
+
+
+    ## Episode End
+    #### Termination
```

Comment:
* termination: If `terminate_when_unhealthy is True` (the default), the environment terminates when the z-position of the humanoid's torso (the height) is no longer contained within `healthy_z_range`.
* truncation: by default, the max duration of an episode is 1000 timesteps